File: src/main/java/com/example/librarymanagement/service/LibraryService.java
package com.example.librarymanagement.service;

import com.example.librarymanagement.model.entity.BookPresence;
import com.example.librarymanagement.model.entity.Library;
import com.example.librarymanagement.model.enums.Availability;

import java.util.List;

public interface LibraryService {
    List<Library> findAll();
    Library getLibraryById(Long id);
    Library createLibrary(Library library);
    Library updateLibrary(Long id, Library updatedLibrary);
    List<BookPresence> getAllBooksByLibraryIdAndAvailability(Long libraryId, Availability availability);
    List<BookPresence> getAllBooksByLibraryId(Long libraryId);
    void deleteLibrary(Long id);
}

File: src/main/java/com/example/librarymanagement/service/LibraryService.kt
package com.example.librarymanagement.service

import com.example.librarymanagement.model.domain.Library

interface LibraryService {
    fun findAll(): List<Library>
    fun getLibraryById(id: String): Library
    fun createLibrary(library: Library): Library
    fun updateLibrary(updatedLibrary: Library): Library
    fun deleteLibraryById(id: String)
}
